File: lib/Model/postModel.dart
import 'package:json_annotation/json_annotation.dart';
part 'postModel.g.dart';

@JsonSerializable()
class PostModel {
  String? username;
  String? fullname;
  String? professor;
  String? influencer;
  String? refproduct;
  String? refbrand;
  String? type;
  String? body;
  double? rating;
  int? comment;
  int? favorite;
  String? forwho;
  DateTime? createdAt;
  DateTime? updatedAt;
  List? coverImage;
  PostModel({this.body,
             this.comment,
             this.favorite,
             this.rating,
             this.fullname,
             this.username,
             this.influencer,
             this.professor,
             this.coverImage,
             this.refbrand,
             this.refproduct,
             this.type,
             this.forwho,
             this.createdAt,
             this.updatedAt});

  factory PostModel.fromJson(Map<String, dynamic> json) =>
      _$PostModelFromJson(json);

  Map<String, dynamic> toJson() => _$PostModelToJson(this);
}

File: Backend/models/post.model.js
const mongoose = require("mongoose");




const Schema = mongoose.Schema;

const Post = Schema(
    {
        email: String,
        username:String,
        fullname: String,
        professor: {
              type: String,
              default: "",
            },
        influencer: {
          type: String,
          default: "",
        },
        img: {
          type: String,
          default: "",
        },
        refproduct: {
            type: String,
            default: ""
        },
        type: String,
        refbrand: {
          type: String,
          default: ""
        },
        body: String,
        coverImage: [
            {
                type: String,
                default: ""
            }
        ],
        comment: {
            type: Number,
            default: "0"
        },
        favorite: {
            type: Number,
            default: "0"
        },
        rating: {
            type: Number,
            default: ""
        },
        forwho:{
            type:String,
            default:"Everyone",
        },
        refcomment: [
            {
                type: mongoose.Schema.Types.ObjectId,
                ref: 'Comment'
            }
        ]
    },
    {
        timestamps: true,
    }
);

module.exports = mongoose.model("Post", Post);